vampyr/trees/trees.cpp

```diff
@@ -0,0 +1,244 @@
+#include "pybind11/eigen.h"
+#include "pybind11/functional.h"
+#include "pybind11/pybind11.h"
+#include "pybind11/stl.h"
+
+#include "trees/BoundingBox.h"
+#include "trees/FunctionTree.h"
+#include "trees/MWTree.h"
+#include "trees/MultiResolutionAnalysis.h"
+#include "trees/OperatorTree.h"
+#include "trees/NodeIndex.h"
+#include "trees/MWNode.h"
+#include "trees/BandWidth.h"
+#include "trees/SerialTree.h"
+#include "trees/SerialFunctionTree.h"
+#include "trees/SerialOperatorTree.h"
+#include "trees/TreeIterator.h"
+
+#include "utils/mpi_utils.h"
+
+#include "treebuilders/add.h"
+#include "treebuilders/apply.h"
+#include "treebuilders/grid.h"
+#include "treebuilders/multiply.h"
+#include "treebuilders/project.h"
+
+using namespace mrcpp;
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace vampyr {
+
+void trees(py::module &m) {
+    const auto D = 3;
+   
+    py::class_<BandWidth> bandwidth(m, "BandWidth");
+    bandwidth.def(py::init<int>())
+        .def("clear", &BandWidth::clear)
+        .def("isEmpty", &BandWidth::isEmpty)
+        .def("getDepth", &BandWidth::getDepth)
+        .def("getMaxWidth", &BandWidth::getMaxWidth)
+        .def("setWidth", &BandWidth::setWidth);
+
+
+
+    py::class_<MWTree<D>> mwtree(m, "MWTree");
+    mwtree.def(py::init<MultiResolutionAnalysis<D>>())
+        .def("getSquareNorm", &MWTree<D>::getSquareNorm)
+        .def("calcSquareNorm", &MWTree<D>::calcSquareNorm)
+        .def("clearSquareNorm", &MWTree<D>::clearSquareNorm)
+        .def("getOrder", &MWTree<D>::getOrder)
+        .def("getKp1", &MWTree<D>::getKp1)
+        .def("getKp1_d", &MWTree<D>::getKp1_d)
+        .def("getDim", &MWTree<D>::getDim)
+        .def("getTDim", &MWTree<D>::getTDim)
+        .def("getNNodes", &MWTree<D>::getNNodes)
+        .def("getNEndNodes", &MWTree<D>::getNEndNodes)
+        .def("getNGenNodes", &MWTree<D>::getNGenNodes)
+        .def("getRootScale", &MWTree<D>::getRootScale)
+        .def("getDepth", &MWTree<D>::getDepth)
+        .def("getSizeNodes", &MWTree<D>::getSizeNodes)      
+        .def("mwTransform", py::overload_cast<int, bool>(&MWTree<D>::mwTransform))
+        .def("setName", &MWTree<D>::setName)
+        .def("getRootIndex", py::overload_cast<const Coord<D> &>(&MWTree<D>::getRootIndex, py::const_))
+        //.def("makeNodeTable", py::overload_cast<MWNodeVector<D> &>(&MWTree<D>::makeNodeTable_))
```

Comment:
Can you leave a comment as to why some methods are commented out?

Reply:
The commented lines on this tree.cpp file were resulting on errors on compiling I didn't manage to fix after some many attempts. I'm not completely sure of their importance on overall coding,  so left them to be "fixed" instead of just deleting.